stwcs/updatewcs/astrometry_utils.py

```diff
@@ -31,10 +32,18 @@
 from lxml import etree
 
 from stwcs.wcsutil import headerlet
-from stwcs.updatewcs import utils
 
 import logging
 logger = logging.getLogger('stwcs.updatewcs.astrometry_utils')
+if len(logger.handlers) > 0:
+    for handler in logger.handlers:
+          # add the handlers to the logger
+          # makes sure no duplicate handlers are added
+        if not isinstance(handler, logging.StreamHandler):
```

Comment:
This will not work if there are a couple of `logging.StreamHandler` in `logger.handlers` (e.g., a `sys.stderr` handler is the first one followed by `sys.stdout`). Then this code will still add multiple `sys.stdout` handlers. Moreover, this modifies `logger.handlers` on the fly. Finally, after adding a `sys.stdout` it does not make sense to continue looping. Most importantly, being an instance of `logging.StreamHandler` does not mean it is a handler for `sys.stdout` => a deeper analysis must be performed.